server/app/repository/ApplicationStatusesRepository.java

```diff
@@ -77,4 +64,23 @@ public void createOrUpdateStatusDefinitions(
       transaction.commit();
     }
   }
+
+  /** Finds all {@link ApplicationStatusesModel} associated with the given program */
+  public ImmutableList<ApplicationStatusesModel> lookupAllApplicationStatusesModels(
+      String programName) {
+    ImmutableList<ApplicationStatusesModel> allApplicationStatusModels =
+        database
+            .find(ApplicationStatusesModel.class)
+            .setLabel("GetAllApplicationStatusesModel.findList")
+            .where()
+            .in("program_name", programName)
+            .query()
+            .findList()
+            .stream()
+            .collect(ImmutableList.toImmutableList());
```

Comment:
(question) Is this list being sorted elsewhere? If not it should have a default case-insensitive sort put on it on the most logical text field the user will see. Just file an issue for your self and do it in a separate PR.

Reply:
Done- https://github.com/civiform/civiform/issues/8131